randovania/games/prime1/exporter/patch_data_factory.py

```diff
@@ -755,6 +755,26 @@ def create_game_specific_data(self) -> dict:
                 for area in region.areas:
                     level_data[region.name]["rooms"][area.name]["submerge"] = self.rng.random() < probability
 
+        # Replace vanilla missile blast shields with the new ones
+        if not self.configuration.legacy_mode:
+            for region in regions:
+                for area in region.areas:
+                    for node in area.nodes:
+                        if isinstance(node, DockNode) and node.dock_type not in elevator_dock_types:
+                            if str(node.default_dock_weakness) != "Missile Blast Shield (randomprime)":
```

Comment:
use node.default_dock_weakness.name , not a string cast

Reply:
I need to really get better at using what is available to me :) theme of this PR